vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -820,6 +830,34 @@ def process_weights_after_loading(self, layer: Module) -> None:
             del layer.w13_input_scale
             del layer.w2_input_scale
 
+        if (not self.rocm_aiter_moe_enabled
+                and self.allow_cutlass_block_scaled_grouped_gemm_sm90):
+            device = w13_weight.device
+            layer.ab_strides1 = torch.full((layer.local_num_experts, ),
+                                           layer.hidden_size,
+                                           device=device,
+                                           dtype=torch.int64)
+            layer.c_strides1 = torch.full(
+                (layer.local_num_experts, ),
+                2 * layer.intermediate_size_per_partition,
+                device=device,
+                dtype=torch.int64)
+            layer.ab_strides2 = torch.full(
+                (layer.local_num_experts, ),
+                layer.intermediate_size_per_partition,
+                device=device,
+                dtype=torch.int64)
+            layer.c_strides2 = torch.full((layer.local_num_experts, ),
+                                          layer.hidden_size,
+                                          device=device,
+                                          dtype=torch.int64)
+            layer.w13_weight_scale_inv = Parameter(
+                layer.w13_weight_scale_inv.transpose(1, 2).contiguous(),
+                requires_grad=False)
+            layer.w2_weight_scale_inv = Parameter(
+                layer.w2_weight_scale_inv.transpose(1, 2).contiguous(),
+                requires_grad=False)
```

Comment:
Is it possible to keep the strides and setup code the same between SM90 and SM100 kernels here? (Again, trying to keep the complexity under control.)